src/edge_index.cpp

```diff
@@ -54,78 +55,33 @@ const HighFive::Group targetIndex(const HighFive::Group& h5Root) {
     return h5Root.getGroup(TARGET_INDEX_GROUP);
 }
 
-Selection resolve(const HighFive::Group& indexGroup, const NodeID nodeID) {
-    if (nodeID >= indexGroup.getDataSet(NODE_ID_TO_RANGES_DSET).getSpace().getDimensions()[0]) {
-        // Returning empty set for out-of-range node IDs, to be aligned with SYN2 reader
-        // implementation
-        // TODO: throw a SonataError instead
-        return Selection({});
-    }
```

Comment:
Here. @matz-e

Reply:
Thanks! We don't deal in SYN2 any longer, maybe we should invoke that TODO (@mgeplf ?)

Reply:
I *think* (but I wouldn't my life on it) this is working around a somewhat ambiguous index case: whether or not the an index range points at the last element, or one past it.  I think keeping the behavior, and not throwing an exeption is fine.